File: src/main/java/com/damo/gestionDeStock/model/CommandeClient.java

package com.damo.gestionDeStock.model;

import javax.persistence.*;
import lombok.*;

import java.time.Instant;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
@Entity
@Table(name = "commandeClient")
public class CommandeClient extends  AbstractEntity{

    @Column(name = "code")
    private String code;

    @Column(name = "identreprise")
    private Integer idEntreprise;

    @Column(name = "datecommande")
    private Instant dateCommande;

    @Column(name = "etatCommande")
    private EtatCommande etatCommande;

    @ManyToOne
    @JoinColumn(name = "idclient")
    private Client client;

    @OneToMany(mappedBy = "commandeClient", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<LigneCommandeClient> ligneCommandeClients;

}

File: src/gs-api/src/models/commande-client.ts
/* tslint:disable */
import { Client } from './client';
import { LigneCommandeClient } from './ligne-commande-client';
export interface CommandeClient {
  id?: number;
  creationDate?: number;
  lastModifiedDate?: number;
  code?: string;
  etatCommande?: 'EN_PREPARATION' | 'VALIDEE' | 'LIVREE';
  idEntreprise?: number;
  dateCommande?: number;
  client?: Client;
  lignecommandeClients?: Array<LigneCommandeClient>;
  username?: boolean;
}
